lib/src/ioctl/request.rs

```diff
@@ -0,0 +1,92 @@
+use nix::libc::{self, c_int};
+use nix::poll::{poll, PollFd};
+use std::fs::File;
+use std::os::unix::io::{AsRawFd, RawFd};
+use std::os::unix::prelude::FromRawFd;
+use thiserror::Error;
+
+pub use nix::poll::PollFlags;
+
+#[doc(hidden)]
+mod ioctl {
+    use nix::libc::c_int;
+    nix::ioctl_read!(media_ioc_request_alloc, b'|', 5, c_int);
+    nix::ioctl_none!(media_request_ioc_queue, b'|', 0x80);
+    nix::ioctl_none!(media_request_ioc_reinit, b'|', 0x81);
+}
+
+#[derive(Debug, Error)]
+pub enum RequestError {
+    #[error("Unexpected ioctl error: {0}")]
+    IoctlError(nix::Error),
+    #[error("Unknown poll flag returned")]
+    UnknownPollFlagReturned,
+}
+
+#[derive(Debug)]
+pub struct Request {
```

Comment:
I think a [typestate](http://cliffle.com/blog/rust-typestate/) would do well here, to prevent e.g. repeated calls to queue, or to denote when a request has been polled and is known to be complete: